### DISCUSSION TITLE
sw_timer.h: No such file or directory

### DISCUSSION THREAD

**Discussion by VincentKen** [Lora-net/LoRaMac-node#1108] [Q&A]
We are trying to import this repository in Atmel Studio 7 without cmake.
So far I have been able to add all include paths except for sw_timer.h which is included in the boards/mcu/saml21/hal/src/hal_timer.c
Where can I find this file? Searching for the file in the github repository only shows a result for this include but not the file itself.
I don't have much experience with CMake so I am also not able to figure out how this include is handled with CMake.

**Comment by mluis1** [comment]
The files under `boards/mcu/saml21` directory have been automatically generated using the Atmel tools.
We tried to touch these files as less as possible. The only changes were to adapt the config files to our requirements.

We are able to compile the project with cmake because we do not use that specific file (`boards/mcu/saml21/hal/src/hal_timer.c`).

I order to know which files we use in this project please refer to the following list under https://github.com/Lora-net/LoRaMac-node/blob/master/src/boards/SAMR34/CMakeLists.txt

https://github.com/Lora-net/LoRaMac-node/blob/57806121493588220ab9d06e87201a02a611e99e/src/boards/SAMR34/CMakeLists.txt#L37-L65
